Description: Don't allow network-config
This may add a new wait time for a file that doesn't exist on existing series
so patch it out.

Author: Brett Holman <brett.holman@canonical.com>
Last-Update: 2024-08-02

--- a/cloudinit/sources/DataSourceNoCloud.py
+++ b/cloudinit/sources/DataSourceNoCloud.py
@@ -190,7 +190,7 @@ class DataSourceNoCloud(sources.DataSour
 
             # This could throw errors, but the user told us to do it
             # so if errors are raised, let them raise
-            md_seed, ud, vd, network = util.read_seeded(seedfrom, timeout=None)
+            md_seed, ud, vd, _ = util.read_seeded(seedfrom, timeout=None)
             LOG.debug("Using seeded cache data from %s", seedfrom)
 
             # Values in the command line override those from the seed
@@ -199,7 +199,6 @@ class DataSourceNoCloud(sources.DataSour
             )
             mydata["user-data"] = ud
             mydata["vendor-data"] = vd
-            mydata["network-config"] = network
             found.append(seedfrom)
 
         # Now that we have exhausted any other places merge in the defaults
--- a/cloudinit/util.py
+++ b/cloudinit/util.py
@@ -1012,7 +1012,6 @@ def read_seeded(base="", ext="", timeout
         ud_url = base.replace("%s", "user-data" + ext)
         vd_url = base.replace("%s", "vendor-data" + ext)
         md_url = base.replace("%s", "meta-data" + ext)
-        network_url = base.replace("%s", "network-config" + ext)
     else:
         if features.NOCLOUD_SEED_URL_APPEND_FORWARD_SLASH:
             if base[-1] != "/" and parse.urlparse(base).query == "":
@@ -1021,17 +1020,7 @@ def read_seeded(base="", ext="", timeout
         ud_url = "%s%s%s" % (base, "user-data", ext)
         vd_url = "%s%s%s" % (base, "vendor-data", ext)
         md_url = "%s%s%s" % (base, "meta-data", ext)
-        network_url = "%s%s%s" % (base, "network-config", ext)
     network = None
-    try:
-        network_resp = url_helper.read_file_or_url(
-            network_url, timeout=timeout, retries=retries
-        )
-    except url_helper.UrlError as e:
-        LOG.debug("No network config provided: %s", e)
-    else:
-        if network_resp.ok():
-            network = load_yaml(network_resp.contents)
     md_resp = url_helper.read_file_or_url(
         md_url, timeout=timeout, retries=retries
     )
--- a/tests/unittests/test_util.py
+++ b/tests/unittests/test_util.py
@@ -2498,7 +2498,7 @@ class TestReadOptionalSeed:
                 {
                     "meta-data": {"md": "val"},
                     "user-data": b"ud",
-                    "network-config": {"net": "cfg"},
+                    "network-config": None,
                     "vendor-data": None,
                 },
                 True,
@@ -2553,7 +2553,7 @@ class TestReadSeeded:
         assert found_md == {"key1": "val1"}
         assert found_ud == ud
         assert found_vd == vd
-        assert found_network == {"test": "true"}
+        assert found_network is None
 
     @pytest.mark.parametrize(
         "base, feature_flag, req_urls",
@@ -2562,7 +2562,6 @@ class TestReadSeeded:
                 "http://10.0.0.1/%s?qs=1",
                 True,
                 [
-                    "http://10.0.0.1/network-config?qs=1",
                     "http://10.0.0.1/meta-data?qs=1",
                     "http://10.0.0.1/user-data?qs=1",
                     "http://10.0.0.1/vendor-data?qs=1",
@@ -2573,7 +2572,6 @@ class TestReadSeeded:
                 "https://10.0.0.1:8008/",
                 True,
                 [
-                    "https://10.0.0.1:8008/network-config",
                     "https://10.0.0.1:8008/meta-data",
                     "https://10.0.0.1:8008/user-data",
                     "https://10.0.0.1:8008/vendor-data",
@@ -2584,7 +2582,6 @@ class TestReadSeeded:
                 "https://10.0.0.1:8008",
                 True,
                 [
-                    "https://10.0.0.1:8008/network-config",
                     "https://10.0.0.1:8008/meta-data",
                     "https://10.0.0.1:8008/user-data",
                     "https://10.0.0.1:8008/vendor-data",
@@ -2595,7 +2592,6 @@ class TestReadSeeded:
                 "https://10.0.0.1:8008",
                 False,
                 [
-                    "https://10.0.0.1:8008network-config",
                     "https://10.0.0.1:8008meta-data",
                     "https://10.0.0.1:8008user-data",
                     "https://10.0.0.1:8008vendor-data",
@@ -2606,7 +2602,6 @@ class TestReadSeeded:
                 "https://10.0.0.1:8008?qs=",
                 True,
                 [
-                    "https://10.0.0.1:8008?qs=network-config",
                     "https://10.0.0.1:8008?qs=meta-data",
                     "https://10.0.0.1:8008?qs=user-data",
                     "https://10.0.0.1:8008?qs=vendor-data",
@@ -2645,7 +2640,7 @@ class TestReadSeeded:
         # user-data, vendor-data read raw. It could be scripts or other format
         assert found_ud == "/user-data: 1"
         assert found_vd == "/vendor-data: 1"
-        assert found_network == {"/network-config": 1}
+        assert found_network is None
         assert [
             mock.call(req_url, timeout=5, retries=10) for req_url in req_urls
         ] == m_read.call_args_list
@@ -2675,7 +2670,7 @@ class TestReadSeededWithoutVendorData(he
         self.assertEqual(found_md, {"key1": "val1"})
         self.assertEqual(found_ud, ud)
         self.assertEqual(found_vd, vd)
-        self.assertEqual(found_network, {"test": "true"})
+        self.assertIsNone(found_network)
 
 
 class TestEncode(helpers.TestCase):
